src/System.Private.DataContractSerialization/src/System/Xml/XmlBaseReader.cs

```diff
@@ -832,6 +832,12 @@ public override XmlNodeType MoveToContent()
                 {
                     if (_node.NodeType != XmlNodeType.Text && _node.NodeType != XmlNodeType.CDATA)
                         break;
+
+                    if (_trailByteCount > 0)
+                    {
+                        break;
+                    }
+
```

Comment:
The ndp\cdf\src codebase is the correct one. ndp\fx\src\cdf code was a fork to support some other internal projects at the time.